clients/src/main/java/org/apache/kafka/common/network/ChannelState.java

```diff
@@ -67,20 +67,23 @@
     // AUTHENTICATION_FAILED has a custom exception. For other states,
     // create a reusable `ChannelState` instance per-state.
     public static final ChannelState NOT_CONNECTED = new ChannelState(State.NOT_CONNECTED);
-    public static final ChannelState AUTHENTICATE = new ChannelState(State.AUTHENTICATE);
     public static final ChannelState READY = new ChannelState(State.READY);
     public static final ChannelState EXPIRED = new ChannelState(State.EXPIRED);
     public static final ChannelState FAILED_SEND = new ChannelState(State.FAILED_SEND);
     public static final ChannelState LOCAL_CLOSE = new ChannelState(State.LOCAL_CLOSE);
 
     private final State state;
     private final AuthenticationException exception;
+    private final String remoteAddress;
+
     public ChannelState(State state) {
-        this(state, null);
+        this(state, null, null);
     }
-    public ChannelState(State state, AuthenticationException exception) {
+    
+    public ChannelState(State state, AuthenticationException exception, String remoteAddress) {
```

Comment:
We probably want another constructor `ChannelState(State state, String remoteAddress)` for non-authentication-failure states where we store `remoteAddress`.

Reply:
:+1: